### ISSUE TITLE
`clearTimeout`  when `debounce.flush()` is called

### ISSUE THREAD

**Issue by gilles-crealp** [lodash/lodash#4074]
When you flush a debounce function, the timer is not cleared.

If the timer is active (`timerId !== undefined`), the function `trailingEdge`is call:

https://github.com/lodash/lodash/blob/0843bd46ef805dd03c0c8d804630804f3ba0ca3c/lodash.js#L10378-L10388

Shouldn't be better with this?

``` javascript
function trailingEdge(time) {
  if (timerId !== undefined) {
    clearTimeout(timerId);
    timerId = undefined;
  }
  // Only invoke if we have `lastArgs` which means `func` has been
  // debounced at least once.
  if (trailing && lastArgs) {
    return invokeFunc(time);
  }
  lastArgs = lastThis = undefined;
  return result;
}
```

**Comment by jdalton** [comment]
Hi @gilles-crealp!

If you'd like to create a PR against our `4.17.12-pre` branch that would rock 🤘

**Comment by jdalton** [comment]
Closed by #4078.

**Comment by gilles-crealp** [comment]
Thank you @blikblum.

You're more reactive than me. 😉
